framework/include/functions/PiecewiseConstantFromCSV.h

```diff
@@ -0,0 +1,55 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "Function.h"
+#include "PropertyReadFile.h"
+
+/**
+ * Function which provides a piecewise constant, in space, field from a CSV file
+ */
+class PiecewiseConstantFromCSV : public Function
+{
+public:
+  static InputParameters validParams();
+
+  PiecewiseConstantFromCSV(const InputParameters & parameters);
+
+  /**
+   * Get the value of the function based on the data in the CSV file
+   * \param t The time (unused)
+   * \param pt The point in space (x,y,z)
+   * \return The value of the function
+   */
+  virtual Real value(Real t, const Point & pt) const override;
+
+  /**
+   * Get the time derivative of the function (based on time only)
+   * \param t The time (unused)
+   * \param pt The point in space (x,y,z) (unused)
+   * \return The time derivative of the function at the specified time
+   */
+  virtual Real timeDerivative(Real, const Point &) const override { return 0; };
```

Comment:
It's valid to take the function time derivative, it's just 0.

Reply:
IMO you either have a docstring override and say that it returns 0 or remove the docstring completely because the method is overridden and the one from the Function class is sufficient, because `t` and `pt` are used, it's just always zero

Reply:
I could say it 'returns 0' in the docstring.
The one from the Function class should be gone. I think that was a bad idea to default to 0 so I removed it

Reply:
Just leaving this:
```
virtual Real timeDerivative(Real, const Point &) const override { return 0; };
```

is sufficient.

This docstring from `Function.h` will be used:

```
  /**
   * Get the time derivative of the function
   * \param t The time
   * \param p The point in space (x,y,z)
   * \return The time derivative of the function at the specified time and location
   */
```

which is still valid with the whole piecewise constant thing

Reply:
oh ok. Yeah I agree with that!